.github/workflows/test-changed-firestore.yml

```diff
@@ -28,27 +34,110 @@ jobs:
           cp config/ci.config.json config/project.json
           yarn
       - name: build
-        run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+        id: build
+        run: yarn build:changed firestore | egrep "Skipping all"
+        # Continue when "Skipping all" is not found
+        continue-on-error: true
+      - name: set output
+        # This means "Skipping all" was not found
+        if: steps.build.outcome != 'success'
+        id: set-output
+        run: echo "CHANGED=true" >> "$GITHUB_OUTPUT";
+      - name: Archive build
+        if: ${{ !cancelled() && steps.build.outcome != 'success' }}
+        run: |
+          tar -cf build.tar --exclude="\.git" .
```

Comment:
nit: Change `--exclude="\.git"` to `--exclude=.git` because you don't have to "escape" the `.` character. The pattern given to `--exclude` is a "glob" pattern (see `man 7 glob`) which does _not_ treat `.` specially.

Reply:
Thanks, will address in the next PR.